includes/RemovePIIJob.php

```diff
@@ -0,0 +1,489 @@
+<?php
+
+namespace Miraheze\RemovePII;
+
+use CentralAuthUser;
+use Exception;
+use ExtensionRegistry;
+use GenericParameterJob;
+use Job;
+use MediaWiki\MediaWikiServices;
+use Title;
+use User;
+use UserProfilePage;
+use WikiPage;
+use Wikimedia\AtEase\AtEase;
+
+class RemovePIIJob extends Job implements GenericParameterJob {
+	/** @var string */
+	private $database;
+
+	/** @var string */
+	private $oldName;
+
+	/** @var string */
+	private $newName;
+
+	/**
+	 * @param array $params
+	 */
+	public function __construct( array $params ) {
+		parent::__construct( 'RemovePIIJob', $params );
+
+		$this->database = $params['database'];
+		$this->oldName = $params['oldname'];
+		$this->newName = $params['newname'];
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function run() {
+		$newCentral = CentralAuthUser::getInstanceByName( $this->newName );
+
+		// Invalidate cache before we begin
+		$newCentral->invalidateCache();
+
+		$oldName = User::newFromName( $this->oldName );
+		$newName = User::newFromName( $this->newName );
+
+		$userOldName = $oldName->getName();
+		$userNewName = $newName->getName();
+
+		if ( !$newName ) {
+			$this->setLastError( "User {$userNewName} is not a valid name" );
+
+			return false;
+		}
+
+		$userId = $newName->getId();
+
+		if ( !$userId ) {
+			$this->setLastError( "User {$userNewName} ID equal to 0" );
+
+			return false;
+		}
+
+		$dbw = wfGetDB( DB_MASTER, [], $this->database );
```

Comment:
`DB_PRIMARY`. Jobs don't support DI yet, but when they do this should be replaced with the DBLoadBalancerFactory service